src/model/document/document.ts

```diff
@@ -266,6 +286,39 @@ variableManager: shallowReadonly(
     return element as any
   }
 
+  function getElementsByType<T extends keyof TElements>(typeName: T): QReturnType<TElements[T]['elementType']>[] | undefined {
```

Comment:
Note to self: `QReturnType<TElements[T]['elementType']>` is pretty complicated. I wonder if I could simplify it like this:
https://stackoverflow.com/a/50924506/3492994

Reply:
Hooray, this works